plugin/possession.lua

```diff
@@ -23,6 +23,11 @@ vim.api.nvim_create_autocmd('VimEnter', {
             vim.fn.delete(symlink)
         end
 
+        if vim.fn.argc() > 0 then
+            -- Skip autoload if any files or folders are passed as command line arguments.
+            return
+        end
+
```

Comment:
This is a first pass at https://github.com/jedrzejboczar/possession.nvim/issues/59#issuecomment-2182073613

It doesn't account for data piped to Neovim, such as `ls ./plugin/*.lua | nvim`. Having a look around, it seems like one solution is to set a global in `StdinReadPre` event, and check this value. Ref:

https://github.com/search?q=repo%3AShatur%2Fneovim-session-manager%20started_with_stdin&type=code

https://github.com/rmagatti/auto-session/blob/e5ffe230235c7d35c2f33078b7cd927a4fb579c4/lua/auto-session/init.lua#L941

If you have a better way, let me know.

Another thing to consider with this: if the user passes only a folder, and that folder has a session, should it still autoload?

Reply:
If we're adding argc check than I think we can also add stdin check. I wouldn't think too much about it, if two other session plugins use this approach then just copy it - it looks fine. We can always change it later if we find a better way. I would just use "namespaced" variable name, e.g. `vim.g.possession_has_stdin`.

As for a single folder, I think we shouldn't treat this specially, see [my comment](https://github.com/jedrzejboczar/possession.nvim/issues/59#issuecomment-2184870086). If users want special behaviour for folders then they can implement it themselves by adding an autocommand.

Reply:
Added `stdin` check. No changes to folder passed as argument; autoload does not occur and Neovim will do whatever it is configured to do.